### PR TITLE
reduce plot_text code and add a test

### PR DESCRIPTION
**Pull request by mxmlnkn**
- [ ] needs rebase against `dev` as soon as #91 is merged

Actually I only wanted to reduce the code and add it to #91, but I wasn't 100% sure, so I added some tests and those tests showed that `const char *` versions were missing and so on ... so this is an extra pull request. Here is what the example image will look like:
texts
Note that the commit overlaps a bit with #93 i.e. with `file_exists` in `pngtest.cc`

### CODE REVIEW

**Comment by mxmlnkn** [comment]
texts

File: src/pngwriter.cc [hunk 1]
```diff
@@ -2023,6 +2023,11 @@ void pngwriter::write_png(void)
 ///////////////////////////////////////////
 void pngwriter::plot_text( char * face_path, int fontsize, int x_start, int y_start, double angle, char * text, double red, double green, double blue)
```
Comment (ax3l) [inline src/pngwriter.cc]: I am not sure a non-const `char *` is needed at all. can you just transform the old one?

Comment (ax3l) [inline src/pngwriter.cc]: what we [actually want](https://stackoverflow.com/questions/890535/what-is-the-difference-between-char-const-and-const-char) is a `const char * const` or even better a c++ `const std::string`

File: examples/pngtest.cc [hunk 2]
```diff
@@ -292,7 +307,59 @@ int main()
...
+   int const textwidth = pngwriter::get_text_width_utf8( font_path.c_str(), 16, "0123456789ABC" );
+   stringstream msg;
+   msg << "width = " << textwidth;
+   float const degToRad = 2.0*3.14159/360.0;
+   texts.filledarrow( 50,70, 50+textwidth,70, 10, 30.0*degToRad, 0.0,0.0,0.0 );
+   texts.filledarrow( 50+textwidth,70, 50,70, 10, 30.0*degToRad, 0.0,0.0,0.0 );
+   texts.plot_text( font_path.c_str(), 16, 50, 85, 0, msg.str().c_str(), 1.0, 0.0, 1.0 );
+   texts.plot_text_blend( font_path.c_str(), 16, 150, 30, 80.0*degToRad, "blended and 80 deg.", 0.5, 0.0, 0.0, 0.0 );
+   /* The UCS4 codes were created with:
+    *   `echo -n '<unicode char here if terminal supports i>' | hexdump`
```
Comment (ax3l) [inline examples/pngtest.cc]: supports *it*?

File: examples/pngtest.cc [hunk 3]
```diff
@@ -292,7 +307,59 @@ int main()
...
+    * The output will be something like:
+    *   0000000 b8c9
+    *   0000002
+    * The long numbers are only addresses, and b8c9 has the problem that
+    * freetype seems to want it the exact other way around, meaning:
```
Comment (ax3l) [inline examples/pngtest.cc]: ~~other~~ opposite

File: src/pngwriter.cc (hunk 1 above)
Comment (mxmlnkn) [inline src/pngwriter.cc]: Well I didn't want to change the interface, else linking with this library wouldn't be downwards compatible I think.

File: examples/pngtest.cc (hunk 2 above)
Comment (ax3l) [inline examples/pngtest.cc]: `... | hexdump -C`

File: examples/pngtest.cc (hunk 3 above)
Comment (ax3l) [inline examples/pngtest.cc]: remove, use `hextdump -C`

File: examples/pngtest.cc [hunk 4]
```diff
@@ -292,7 +307,59 @@ int main()
...
+    *   "\xc9\xb8"
+    * @see https://en.wikipedia.org/wiki/Byte_order_mark
+    * The smiley icon gives:
+    *   0000000 98e2 00ba
```
Comment (ax3l) [inline examples/pngtest.cc]:
```bash
$ echo -n "☺" | hexdump -C                                               
00000000  e2 98 ba                                          |...|
00000003
```

Comment (ax3l) [inline examples/pngtest.cc]: or `... | hd`

File: src/pngwriter.cc (hunk 1 above)
Comment (ax3l) [inline src/pngwriter.cc]: you are already making it ABI and API incompatible with the other changes, it is going into `0.6.0` and will be at least compatible on the API with the `std::string` constructor.

Comment (ax3l) [inline src/pngwriter.cc]: ah I see you try to keep the API the same in the other PR. Well, then it still adds to the API so I leave the label in for the change log.

still, I am not a big fan of adding more c-style string APIs. but this can be refactored in an independent PR.

File: examples/pngtest.cc (hunk 4 above)
Comment (mxmlnkn) [inline examples/pngtest.cc]: ah, interesting, works for me too, thanks.
